common/djangoapps/third_party_auth/tasks.py

```diff
@@ -36,7 +37,7 @@ def fetch_saml_metadata():
         num_failed: Number of providers that could not be updated
         num_total: Total number of providers whose metadata was fetched
     """
-    if not SAMLConfiguration.is_enabled():
+    if not SAMLConfiguration.is_enabled(get_current_site_default()):
```

Comment:
Oops! Yes, you are right, what we really need to do is run this for any SAMLProviderConfig with a matching SAMLConfiguration enabled for it's Site.

I've made the fix and pushed.

By the way, I took a look and I don't think this is run at startup, it's run whenever a SAMLConfiguration is updates, and also the "saml pull" management command.